### ISSUE TITLE
Too many GET requests

### ISSUE THREAD

**Issue by audreywl** [olinlibrary/abe-web#43]
It happens every time we switches views. Fixing this is only relevant if we stick with FullCalendar

**Comment by minhkhang1795** [comment]
Hi,

My name is Khang. I've just found out this cool project recently from the orientation folder for first-year students. I found a bug and believe that it may be related to this thread.

**Bug Description:** When we switch the views/tabs (from month to week or to day). The events are not updated correctly.

**How to Reproduce:**
1. Go to [https://events.olin.build/calendar/academic,burstthebubble,featured,food,holidays](https://events.olin.build/calendar/academic,burstthebubble,featured,food,holidays)
2. In "Tags", click **"None"**. Then, click the **"Food"** tag only.
3. Currently, we are at the "Month" tab. See that on Wed Feb 13, 2017, we have an event at 12:30pm Community Breakfast (sometimes this event turns out to be at 8:30 am or something, but I only saw that once, so I'm not sure about this other issue). Now switch to other tabs like Week and Day to find the event on that day. We probably wouldn't find it. 
4. Also, there is an "All-day" row in the Week tab and the Day tab. We can't find the event in those rows either.

**Screenshots:**
![one](http://i.imgur.com/9eNBAEF.png)

![two](http://i.imgur.com/NVkq1Mr.png)

![three](http://i.imgur.com/J9rF5ot.png)

I really love what you guys are doing and you have my full support!!! 
Best,
Khang Vu

**Comment by kylecombes** [comment]
Hi @minhkhang1795,

Thanks for your message! Yes, I have also noticed there is some undesired behavior switching between views. FullCalendar (the jQuery plugin we are currently mashing with an otherwise React-based app) has given us a fair amount of grief over the past couple of months, and I'm currently in the process of designing [a React-based replacement](https://github.com/kylecombes/ultra-responsive-calendar) that meets our needs. (There is also the [Dayz](https://github.com/nathanstitt/dayz) React package, but that doesn't quite give us the flexibility we're after.)

If you find anything other bugs, please don't hesitate to let us know by opening a new issue! Also, if you have any interest helping out with this project, we're always eager to have someone new join us. No React/JavaScript/Web-dev experience required (we learned most of this over the summer).

Enjoy the rest of orientation, and I look forward to meeting you soon!

Kyle

**Comment by minhkhang1795** [comment]
Hi, 

I literally have no experience in this stuff, but I'd love to join. I'd love to see if we could make a calendar for professors and students to communicate with one another effectively as a whole unique system. Today, I got the class schedule and I had to add them all to my outlook calendar manually, which was a very tedious, boring task.

Still, I need to take some time figuring out what I should do in my first semester.  So I look forward to meeting you guys someday.

Best.
Khang Vu
